integration/k8s_fixture_test.go

```diff
@@ -168,8 +179,125 @@ func (f *k8sFixture) ClearNamespace() {
 	f.ClearResource("services")
 }
 
+func (f *k8sFixture) getOldKubeConfig() {
+	usr, _ := user.Current()
+	dir := usr.HomeDir
+	// NOTE(dmiller): this assumes that your kube config exists in the default location, and is the one currently being used.
+	// This is true in CI and on my machine, but is not universally true for anyone who might want to run this test on their machine.
+	kubeConfigPath := filepath.Join(dir, ".kube", "config")
+
+	old, err := ioutil.ReadFile(kubeConfigPath)
+	if err != nil {
+		f.t.Fatalf("Error reading KUBECONFIG: %v", err)
+	}
+	info, err := os.Stat(kubeConfigPath)
+	if err != nil {
+		f.t.Fatalf("Error stat'ing KUBECONFIG: %v", err)
+	}
+
+	f.oldKubeConfigPath = kubeConfigPath
+	f.oldKubeConfigPerms = info.Mode()
+	f.oldKubeConfig = string(old)
+}
+
+func (f *k8sFixture) maybeRestoreOldKubeConfig() {
+	if f.oldKubeConfig != "" {
+		err := ioutil.WriteFile(f.oldKubeConfigPath, []byte(f.oldKubeConfig), f.oldKubeConfigPerms)
+		if err != nil {
+			f.t.Fatalf("Error restoring old KUBECONFIG: %v", err)
+		}
+	}
+}
+
+func (f *k8sFixture) createUser() {
+	outWriter := bytes.NewBuffer(nil)
+	cmd := exec.CommandContext(f.ctx, "kubectl", "apply", "-f", "access.yaml")
+	cmd.Stdout = outWriter
+	cmd.Stderr = outWriter
+	cmd.Dir = packageDir
+	err := cmd.Run()
+	if err != nil {
+		f.t.Fatalf("Error creating user: %v. Logs:\n%s", err, outWriter.String())
+	}
+}
+
+func (f *k8sFixture) getSecrets() {
+	outWriter := bytes.NewBuffer(nil)
+	cmdStr := `kubectl get secrets -n tilt-integration -o json | jq -r '.items[] | select(.metadata.name | startswith("tilt-integration-user-token-")) | .data.token'`
+	cmd := exec.CommandContext(f.ctx, "bash", "-c", cmdStr)
+	cmd.Stderr = outWriter
+	cmd.Dir = packageDir
+	token, err := cmd.Output()
+	if err != nil {
+		f.t.Fatalf("Error getting secrets: %v. Cmd: %s", err, cmdStr)
+	}
+
+	cmdStr = `kubectl get secrets -n tilt-integration -o json | jq -r '.items[] | select(.metadata.name | startswith("tilt-integration-user-token-")) | .data["ca.crt"]'`
+	cmd = exec.CommandContext(f.ctx, "bash", "-c", cmdStr)
+	cmd.Stderr = outWriter
+	cmd.Dir = packageDir
+	cert, err := cmd.Output()
+	if err != nil {
+		f.t.Fatalf("Error getting secrets: %v. Cmd: %s", err, cmdStr)
+	}
+
+	f.token = string(token)
+	f.cert = string(cert)
+}
+
+func (f *k8sFixture) SetRestrictedCredentials() {
+	f.getSecrets()
+
+	err := ioutil.WriteFile("/tmp/cert", []byte(f.cert), 0644)
```

Comment:
are you aware of client-certificate-data so you can embed the data in the kubeconfig file?

Reply:
Yes, but you can't specify that via `kubectl config set-credentials` and I'm trying to avoid having to parse/write a kubeconfig.

I don't think I can avoid it though, tragically.

Reply:
have you tried `kubectl config set`? its instructions explicitly mention certificate-authority-data

Reply:
Aha, no! Thank you for the tip